new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java

```diff
@@ -0,0 +1,1258 @@
+
+    assertTrue(crdPatched, "patchDomainCustomResource(adminShutdown) failed");
+   
+    checkPodDeleted(adminServerPodName, domainUid, domainNamespace);
+    logger.info("AdminServer shutdown success");
+
+    logger.info("Check managed server pods are not affected");
+    DateTime dynTs2 = getPodCreationTime(domainNamespace, dynamicServerPodName);
+    DateTime cfgTs2 = getPodCreationTime(domainNamespace, configServerPodName);
+
+    assertTrue(
+        dynTs2.withTimeAtStartOfDay().isEqual(dynTs.withTimeAtStartOfDay()), 
+        "Dynamic managed server pod creation time must be same");
+    assertTrue(
+        cfgTs2.withTimeAtStartOfDay().isEqual(cfgTs.withTimeAtStartOfDay()), 
+        "Configured managed server pod creation time must be same");
+
+    patchStr = null;
+    patchStr = new StringBuffer("[{");
+    patchStr.append("\"op\": \"replace\",")
+        .append(" \"path\": \"/spec/adminServer/serverStartPolicy\",")
+        .append(" \"value\":  \"IF_NEEDED\"")
+        .append(" }]");
+    logger.info("AdminServer restart patch string: {0}", patchStr);
+
+    patch = new V1Patch(new String(patchStr));
+    crdPatched = false;
+    crdPatched = assertDoesNotThrow(() ->
+            patchDomainCustomResource(domainUid, domainNamespace, patch, "application/json-patch+json"),
+        "patchDomainCustomResource(restart) failed");
+
+    assertTrue(crdPatched, "patchDomainCustomResource(restart) failed");
+    logger.info("Check admin service {0} is created in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkServiceExists(adminServerPodName, domainNamespace);
+    logger.info("Wait for admin server pod {0} to be ready in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodReady(adminServerPodName, domainUid, domainNamespace);
+    logger.info("AdminServer restart success");
+  }
+
+  /**
+   * Stop a configured cluster by patching the resource definition with 
+   *  spec/clusters/1/serverStartPolicy set to NEVER.
+   * Make sure that only server(s) in the configured cluster are stopped. 
+   * Make sure that server(s) in the dynamic cluster are in RUNNING state. 
+   * Restart the cluster by patching the resource definition with 
+   *  spec/clusters/1/serverStartPolicy set to IF_NEEDED.
+   * Make sure that servers in the configured cluster are in RUNNING state. 
+   */
+  @Test
+  @DisplayName("Restart the configured cluster with serverStartPolicy")
+  public void testConfigClusterRestart() {
+
+    String configServerPodName = domainUid + "-config-cluster-server1";
+    String dynamicServerPodName = domainUid + "-managed-server1";
+
+    DateTime dynTs = getPodCreationTime(domainNamespace, dynamicServerPodName);
+
+    checkServiceExists(configServerPodName, domainNamespace);
+    checkPodReady(configServerPodName, domainUid, domainNamespace);
+    logger.info("(BeforePatch) configured cluster managed server is RUNNING");
+
+    StringBuffer patchStr = null;
```

Comment:
duplicated code, create a method and use every where

Reply:
done